### DISCUSSION TITLE
Add transit layer in <Map/>

### DISCUSSION THREAD

**Discussion by MicheleFatticcioni** [visgl/react-google-maps#440] [Q&A] [answered]
Hello, I'm working with '@vis.gl/react-google-maps' and AdvancedMarker.
Here is a piece of my implementation:
![image](https://github.com/visgl/react-google-maps/assets/77053757/824057dc-a11b-4e77-b689-a0ea9e96c635)

I'm trying to add a transit layer to my map, but I can't find a way to do it.

**Comment by usefulthink** [comment, marked as answer]
You can roughly follow this guide here: https://visgl.github.io/react-google-maps/docs/guides/interacting-with-google-maps-api#hooks

For example, you create a `TransitLayer` component and use the `useMap` / `useEffect` construct to add the transitLayer:

```tsx
const TransitLayer = () => {
  const map = useMap();
  useEffect(() => {
    if (!map) return;

    const transitLayer = new google.maps.TransitLayer();
    transitLayer.setMap(map);

    return () => transitLayer.setMap(null);
  }, [map]);

  return <></>;
}
```

(since this component doesn't need to render anything, it would look almost the same as a custom hook `useTransitLayer` if you prefer it that way)

**Reply by MicheleFatticcioni** [reply]
Thank you! You saved my day

**Reply by usefulthink** [reply]
happy to help!
